src/filetypes.rs

```diff
@@ -15,10 +15,11 @@ pub struct FileTypes {
 
 impl FileTypes {
     pub fn should_ignore(&self, entry: &walk::DirEntry) -> bool {
+        let is_symlink = entry.is_symlink();
         if let Some(ref entry_type) = entry.file_type() {
-            (!self.files && entry_type.is_file())
-                || (!self.directories && entry_type.is_dir())
-                || (!self.symlinks && entry_type.is_symlink())
+            (!self.files && entry_type.is_file() && !is_symlink)
```

Comment:
It might be better for this (and the following line) to be something like:

```suggestion
            (!self.files && entry_type.is_file() && !(self.symlinks && is_symlink))
```
(maybe factoring that out self.symlinks && is_symlink into a variable?)

so that if you specify only `--type file` it doesn't include symlinks to directories and vice versa.